### PR TITLE
Xcode shortcuts

### PR DESCRIPTION
**Pull request by marwendoukh**
# Goodie Pull Request Template

**What does your Instant Answer do?**
Xcode shortcuts

**What problem does your Instant Answer solve (Why is it better than organic links)?**
show  Xcode shortcuts without a mouse click

**What is the data source for your Instant Answer? (Provide a link if possible)**
https://developer.apple.com/library/ios/documentation/IDEs/Conceptual/xcode_help-command_shortcuts/MenuCommands/MenuCommands014.html

**Why did you choose this data source?**
officiel apple website

**Are there any other alternative (better) data sources?**
http://www.git-tower.com/blog/xcode-cheat-sheet/

http://spin.atomicobject.com/2014/03/23/xcode-keyboard-shortcuts/

**What are some example queries that trigger this Instant Answer?**
xcode cheat sheet

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
developpers

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
https://duck.co/ideas/idea/6239/xcode-cheat-sheet

**Which existing Instant Answers will this one supercede/overlap with?**

**Are you having any problems? Do you need our help with anything?**
screenshot from 2015-08-11 11 11 48

**Where did you hear about DuckDuckHack? (For first time contributors)**

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**

## Checklist
Please place an 'X' where appropriate.

```
- [X] Added metadata and attribution information (https://duck.co/duckduckhack/metadata)
- [] Wrote test file and added to t/ directory (https://duck.co/duckduckhack/goodie_tests)
- [X] Verified that instant answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
- [X] Verified that instant answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
- [] Tested cross-browser compatibility

    Please let us know which browsers/devices you've tested on:
    - Windows 8
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] IE 10

    - Windows 7
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] IE 8
        - [] IE 9
        - [] IE 10

    - Windows XP
        - [] IE 7
        - [] IE 8

    - Mac OSX
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] Safari

    - iOS (iPhone)
        - [] Safari Mobile
        - [] Google Chrome
        - [] Opera

    - iOS (iPad)
        - [] Safari Mobile
        - [] Google Chrome
        - [] Opera

    - Android
        - [] Firefox
        - [] Native Browser
        - [] Google Chrome
        - [] Opera
    - Linux
        - [X] Firefox
        - [] Native Browser
        - [] Google Chrome
        - [] Opera

```

-----
IA Page: https://duck.co/ia/view/xcode_cheat_sheet

### CODE REVIEW

File: share/goodie/cheat_sheets/xcode.json [hunk 1]
```diff
@@ -0,0 +1,376 @@
+/**
+* Created with DuckDuckHack Xcode shortcuts.
+* User: marwendoukh
+* Date: 2015-08-11
+* Time: 10:33 AM
+*/
+
+{
+    // Required; must match the id of the cheat sheet's IA page
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/xcode.json]: These comments can go

**Comment by moollaza** [comment]
@abeyang any thoughts on this CheatSheet? It's using the OSX icons for Command, Control Shift, and Option.

**Comment by moollaza** [comment]
@marwendoukh thanks a lot for contributing! We really appreciate it :smile: 

The content and syntax LGTM. Just want @abeyang to see how it looks using the ASCII icons. So far we've been using the words "Command", "Option", "Shift", etc.

**Comment by abeyang** [comment]
@marwendoukh Man, I would love to use these glyphs, but for now, can we just use the words Cmd/Option/etc? Just FYI, this is something we're trying to work on internally, but rather than using ASCII symbols (which looks hokey), we'd like to design our own glyphs.

(It sure does look a lot neater, though.)

**Comment by marwendoukh** [comment]
Ping @moollaza  @abeyang

File: share/goodie/cheat_sheets/xcode.json [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+    "section_order": [  
+        "Xcode menu command",
+        "File menu command",
+        "Edit menu command",
+        "View menu command",
+        "Navigate menu command",
+        "Editor menu command", 
+        "Product menu command",
+        "Window menu command",
+        "Help menu command",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/xcode.json]: Whoops, there's a trailing comma here! Please remove it as it's causing an error. Make sure you test your changes before submitting them so you're sure they work :+1:

**Comment by moollaza** [comment]
@marwendoukh this is *juust* about done. Could you pull in the changes from upstream and `git mv` the json file into the new /json/ subdir in `/share/goodies/cheat_sheets` as well?

Thanks a lot!

**Comment by marwendoukh** [comment]
I corrected the syntax error
What's next ?

**Comment by moollaza** [comment]
@marwendoukh thanks a lot! This LGTM :+1:

**Comment by moollaza** [comment]
@marwendoukh congrats! It's live: https://duckduckgo.com/?q=xcode+cheat+sheet&ia=cheatsheet

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's, tackle open issues or even submit more Instant Answers!

**Pull request merged**